src/ipmi.c

```diff
@@ -200,56 +202,66 @@ static void sensor_read_handler(ipmi_sensor_t *sensor, int err,
-
+static void sensor_get_name(ipmi_sensor_t *sensor, char *buf, int buf_len) {
   char buffer[DATA_MAX_NAME_LEN] = {0};
-  const char *entity_id_string;
-  char sensor_name[DATA_MAX_NAME_LEN];
-  char *sensor_name_ptr;
-  int sensor_type;
-  const char *type;
   ipmi_entity_t *ent = ipmi_sensor_get_entity(sensor);
-
-  sensor_id = ipmi_sensor_convert_to_id(sensor);
+  const char *entity_id_string = ipmi_entity_get_entity_id_string(ent);
+  char sensor_name[DATA_MAX_NAME_LEN] = "";
+  char *sensor_name_ptr;
 
   ipmi_sensor_get_name(sensor, buffer, sizeof(buffer));
   buffer[sizeof(buffer) - 1] = 0;
 
-  entity_id_string = ipmi_entity_get_entity_id_string(ent);
-
-  if (entity_id_string == NULL)
-    sstrncpy(sensor_name, buffer, sizeof(sensor_name));
-  else
+  if (entity_id_string != NULL && strlen(buffer))
     ssnprintf(sensor_name, sizeof(sensor_name), "%s %s", buffer,
               entity_id_string);
+  else if (entity_id_string != NULL)
+    sstrncpy(sensor_name, entity_id_string, sizeof(sensor_name));
+  else
+    sstrncpy(sensor_name, buffer, sizeof(sensor_name));
 
-  sstrncpy(buffer, sensor_name, sizeof(buffer));
-  sensor_name_ptr = strstr(buffer, ").");
-  if (sensor_name_ptr != NULL) {
-    /* If name is something like "foo (123).bar",
-     * change that to "bar (123)".
-     * Both, sensor_name_ptr and sensor_id_ptr point to memory within the
-     * `buffer' array, which holds a copy of the current `sensor_name'. */
-    char *sensor_id_ptr;
-
-    /* `sensor_name_ptr' points to ").bar". */
-    sensor_name_ptr[1] = 0;
-    /* `buffer' holds "foo (123)\0bar\0". */
-    sensor_name_ptr += 2;
-    /* `sensor_name_ptr' now points to "bar". */
-
-    sensor_id_ptr = strstr(buffer, "(");
-    if (sensor_id_ptr != NULL) {
-      /* `sensor_id_ptr' now points to "(123)". */
-      ssnprintf(sensor_name, sizeof(sensor_name), "%s %s", sensor_name_ptr,
-                sensor_id_ptr);
+  if (strlen(buffer)) {
+    sstrncpy(buffer, sensor_name, sizeof(buffer));
+    sensor_name_ptr = strstr(buffer, ").");
+    if (sensor_name_ptr != NULL) {
+      /* If name is something like "foo (123).bar",
+       * change that to "bar (123)".
+       * Both, sensor_name_ptr and sensor_id_ptr point to memory within the
+       * `buffer' array, which holds a copy of the current `sensor_name'. */
+      char *sensor_id_ptr;
+
+      /* `sensor_name_ptr' points to ").bar". */
+      sensor_name_ptr[1] = 0;
+      /* `buffer' holds "foo (123)\0bar\0". */
+      sensor_name_ptr += 2;
+      /* `sensor_name_ptr' now points to "bar". */
+
+      sensor_id_ptr = strstr(buffer, "(");
+      if (sensor_id_ptr != NULL) {
+        /* `sensor_id_ptr' now points to "(123)". */
+        ssnprintf(sensor_name, sizeof(sensor_name), "%s %s", sensor_name_ptr,
+                  sensor_id_ptr);
+      }
+      /* else: don't touch sensor_name. */
     }
-    /* else: don't touch sensor_name. */
   }
-  sensor_name_ptr = sensor_name;
+
+  assert(buf != NULL);
```

Comment:
Please don't do argument checking with `assert()`. Instead, add something like this to the beginning of the function.

```c
if ((buf == NULL) || (buf_len == 0)) {
  return;
}
```